src/modules/NtpClient.rb

```diff
@@ -795,24 +797,37 @@ def write_and_update_policy
       success
     end
 
-    # Enable or disable ntp service depending on @run_service value
+    # Enable or disable chrony services depending on @run_service value
+    # "chrony-wait" service has also to be handled in order to ensure that
+    # "chronyd" is working correctly and do not depend on the network status.
     #
-    # * When disabling, it also stops the service.
-    # * When enabling, it tries to restart the service unless it's in write
+    # * When disabling, it also stops the services.
+    # * When enabling, it tries to restart the services unless it's in write
     #   only mode.
     def check_service
-      adjusted = @run_service ? Service.Enable(@service_name) : Service.Disable(@service_name)
-
-      # error report
-      Report.Error(Message.CannotAdjustService("NTP")) unless adjusted
+      if @run_service
```

Comment:
this method contains two separate workflows enclosed in two ```if @run_service``` branches. Moreover first of them do not contain else part. It can be rewritten to avoid one level of nesting e.g. like
```
def check_service
  if !@run_service
    Service.Stop(@service_name)
    Service.Stop(@wait_service_name)
    return
  end
....
end
````
which seems to be more readable for me.